src/seedu/addressbook/data/person/Email.java

```diff
@@ -21,13 +21,13 @@
      *
      * @throws IllegalValueException if given email address string is invalid.
      */
-    public Email(String email, boolean isPrivate) throws IllegalValueException {
+    public Email(String emailValue, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        email = email.trim();
```

Comment:
This is the problematic line. i.e. changing the value of the parameter, which is a discouraged practice.
The fix is,
`String trimmedEmail = email.trim();`

Adding the `Value` suffix to the parameter doesn't solve the problem if you reassign it later. Furthermore, that additional suffix makes it inconsistent with other similar places.